content/en/docs/rules/style-guide.md

```diff
@@ -0,0 +1,164 @@
+linktitle: Style Guide of Falco Rules
+weight: 95
+---
+
+## Style Guide
+
+Before diving in, read the sections on Falco rules [basics](/docs/rules/basic-elements/) and [condition syntax](/docs/rules/conditions/). Also, check out existing [upstream rules](https://github.com/falcosecurity/rules/blob/main/rules/falco_rules.yaml) for best practices in writing rules.
+
+In addition, the resources under [references/rules](/docs/reference/rules/) provide complementary information. We highly recommend regularly revisiting each guide to stay up-to-date with the latest advancements of Falco.
+
+The order of keys should be:
+
+```yaml
+- rule: 
+  desc:
+  condition:
+  output:
+  priority:
+  tags:
+
+ ... other keys if applicable in no particular order
+
+```
+
+### Naming
+
+Choose a concise title that summarizes the essence of the rule's purpose.
+
+### Description
+
+Aligning with Falco's rules maturity and adoption [framework](https://github.com/falcosecurity/rules/blob/main/proposals/20230605-rules-adoption-management-maturity-framework.md#rules-maturity-framework), it is encouraged to not just include a longer description of what the rule detects but also to give advice on how to tune this rule and reduce possible noise. If applicable, elaborate on how to correlate the rule with other rules or data sources for incident response. However, keep them concise.
+
+### Condition Syntax
+
+These recommendations prioritize performance impact while maintaining a consistent style for better understanding and easier customization. This approach ensures more manageable maintenance of the rules in the long run.
+
+We explain the high level principles using example rules or snippets.
+
+- Each upstream Falco rule must include an `evt.type` filter.
+- Prioritize the `evt.type` filter first. Falco buckets filters per `evt.type` for efficient rules matching through applying the rule's Abstract Syntax Tree (AST) to relevant event types only. A nice side effect is better readability as well.
+- To maintain performance, avoid mixing unrelated event types in one rule. Typically, only variants should be mixed together, for example: `evt.type in (open, openat, openat2)`.
+- The best practice and requirement for upstream rules are to only define positive `evt.type` expressions. Using `evt.type!=open`, for example, would imply each of the [supported syscalls](https://github.com/falcosecurity/libs/blob/master/driver/report.md), resulting in a significant performance penalty. For more information, read the [Adaptive Syscalls Selection in Falco](https://falco.org/blog/adaptive-syscalls-selection/) blog post.
+- After the `evt.type` filter, place your mainly positive filters to efficiently eliminate the most events step by step. An exception to this rule is the `container` macro, which can quickly eliminate many events. Therefore, the guiding principle of "divide and conquer" commonly used in database query recommendations, also applies to Falco's filter statements.
```

Comment:
Those are nice! However, they apply only to the `syscall` data source. Do we want to mention that?